sdk/core/Azure.Core.Experimental/tests/Variant/StoringBoolean.cs

```diff
@@ -0,0 +1,169 @@
+        public void BooleanImplicit(bool testValue)
+        {
+            Value value;
+            using (MemoryWatch.Create)
+            {
+                value = testValue;
+            }
+
+            Assert.AreEqual(testValue, value.As<bool>());
+            Assert.AreEqual(typeof(bool), value.Type);
+
+            bool? source = testValue;
+            using (MemoryWatch.Create)
+            {
+                value = source;
+            }
+            Assert.AreEqual(source, value.As<bool?>());
+            Assert.AreEqual(typeof(bool), value.Type);
+        }
+
+        [TestCase(true)]
+        [TestCase(false)]
+        public void BooleanCreate(bool testValue)
+        {
+            Value value;
+            using (MemoryWatch.Create)
+            {
+                value = Value.Create(testValue);
+            }
+
+            Assert.AreEqual(testValue, value.As<bool>());
+            Assert.AreEqual(typeof(bool), value.Type);
+
+            bool? source = testValue;
+
+            using (MemoryWatch.Create)
+            {
+                value = Value.Create(source);
+            }
+
+            Assert.AreEqual(source, value.As<bool?>());
+            Assert.AreEqual(typeof(bool), value.Type);
+        }
+
+        [TestCase(true)]
+        [TestCase(false)]
+        public void BooleanInOut(bool testValue)
+        {
+            Value value;
+            bool success;
+            bool result;
+
+            using (MemoryWatch.Create)
+            {
+                value = new(testValue);
+                success = value.TryGetValue(out result);
+            }
+
+            Assert.True(success);
+            Assert.AreEqual(testValue, result);
+
+            Assert.AreEqual(testValue, value.As<bool>());
+            Assert.AreEqual(testValue, (bool)value);
+        }
+
+        [TestCase(true)]
+        [TestCase(false)]
+        public void NullableBooleanInBooleanOut(bool? testValue)
+        {
+            bool? source = testValue;
+            Value value;
+            bool success;
+            bool result;
+
+            using (MemoryWatch.Create)
+            {
+                value = new(source);
+                success = value.TryGetValue(out result);
+            }
+
+            Assert.True(success);
+            Assert.AreEqual(testValue, result);
+
+            Assert.AreEqual(testValue, value.As<bool>());
+
+            Assert.AreEqual(testValue, (bool)value);
+        }
+
+        [TestCase(true)]
+        [TestCase(false)]
+        public void BooleanInNullableBooleanOut(bool testValue)
+        {
+            bool source = testValue;
+            Value value = new(source);
+            bool success = value.TryGetValue(out bool? result);
+            Assert.True(success);
+            Assert.AreEqual(testValue, result);
+
+            Assert.AreEqual(testValue, (bool?)value);
+        }
+
+        [TestCase(true)]
+        [TestCase(false)]
+        public void BoxedBoolean(bool testValue)
```

Comment:
Value can store any [literally any] value. This test tests that it can store boxed bool and that the semantics of such scenario are what's expected. For example, that you can retrieve a bool when the value stored is a boxed bool, as opposed to having to retrieve an object and then manually unbox it.